spec/ic.did

```diff
@@ -19,7 +18,7 @@ type definite_canister_settings = record {
 type http_header = record { name: text; value: text };
 
 type http_response = record {
-  status: nat;
+  status: nat64;
```

Comment:
The team thinks that this should be changed to nat64 in the spec as it does not make sense to have an arbitrary precision number for representing HTTP status codes.

Reply:
Probably not that important, but this goes against the advice by @rossberg to not use fixed-width numbers unless you do bit-fiddling or something like that.

Also note that `nat` is actually shorter on the wire than `nat64` for small values.

And even if, why `nat64` and not `nat16` or `nat32`?

Reply:
I wonder if this push to use `nat64` is mostly an artifact of the Rust bindings being designed in a way that makes the use of `nat` more inconvenient. This could probably be fixed by adding a newtype around `i64` to the Rust library that encodes/decodes to `nat`, and errors out when the incoming number is too large. Then Rust code can interact with `nat` with very little extra work whenever it wants to reject very large values anyways. WDYT, @chenyan-dfinity?

Reply:
`i128`/`u128` already maps to `int`/`nat` in Candid. The question is whether we want to optimize space for messages or the Rust memory.

We could define a newtype for `i64`, but people may still choose to ignore it. Just like we already have `candid::Int` and `candid::Nat` defined. Maybe we need an option to map all Rust number types to `nat`/`int` on the wire?

Reply:
@nomeata : Thanks for your thoughts about this and the reference to Andreas Rossberg's opinion. So you would advise to keep it as a `nat`?

Reply:
Yes, I guess just use `i128` in rust and not worry any further :-)

Reply:
@nomeata, @chenyan-dfinity :Sounds good! Thanks for your helpful inputs here!